ui/resolution_selection.js

```diff
@@ -95,7 +99,22 @@ shaka.ui.ResolutionSelection = class extends shaka.ui.SettingsMenu {
     // If there is a selected variant track, then we filter out any tracks in
     // a different language.  Then we use those remaining tracks to display the
     // available resolutions.
-    const selectedTrack = tracks.find((track) => track.active);
+    let selectedTrack = tracks.find((track) => track.active);
+    if (this.flag ==1 && tracks.find((track) =>
+      track.height ==
+      Number((this.currentSelection.textContent.toString()).slice(0, -1))) !==
+      undefined) {
+      selectedTrack = tracks.find((track) => track.height ==
+      Number((this.currentSelection.textContent.toString()).slice(0, -1)));
```

Comment:
This condition is not ideal and confusing to understand.  It would be better to somehow use the track list to select the track.  For example, you could manually fire the `click` event for the currently-selected resolution.  Or you could track the mapping between the resolution list and the track associated with it.